lib/Frontend/CompilerInvocation.cpp

```diff
@@ -844,6 +844,7 @@ static bool ParseSILArgs(SILOptions &Opts, ArgList &Args,
   Opts.RemoveRuntimeAsserts |= Args.hasArg(OPT_RemoveRuntimeAsserts);
 
   Opts.EnableARCOptimizations |= !Args.hasArg(OPT_disable_arc_opts);
```

Comment:
This seems like an obvious bug on line 846. I am going to fix this in a separate PR.